pkg/resource/sensitive.go

```diff
@@ -65,25 +68,16 @@ type SecretClient interface {
 // GetConnectionDetails returns connection details including the sensitive
 // Terraform attributes and additions connection details configured.
 func GetConnectionDetails(attr map[string]interface{}, tr Terraformed, cfg *config.Resource) (managed.ConnectionDetails, error) {
-	conn, err := GetSensitiveAttributes(attr, tr.GetConnectionDetailsMapping(), tr.GetTerraformResourceIDField())
+	conn, err := GetSensitiveAttributes(attr, tr.GetConnectionDetailsMapping())
 	if err != nil {
 		return nil, errors.Wrap(err, "cannot get connection details")
 	}
 
-	var custom map[string][]byte
-	// TODO(turkenh): Once we have automatic defaulting, remove this if check.
```

Comment:
Is the idea here that we will always have a non-zero `AdditionalConnectionDetailsFn` because of the defaulting? I would still prefer to keep this check so that a nil `AdditionalConnectionDetailsFn` continues to behave as a noop.

Reply:
See my comment in https://github.com/crossplane-contrib/terrajet/pull/132#discussion_r751677581